packages/@aws-cdk/aws-apigatewayv2/lib/http/integrations/http-private.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  HttpConnectionType,
+  HttpIntegrationType,
+  HttpRouteIntegrationBindOptions,
+  HttpRouteIntegrationConfig,
+  IHttpRouteIntegration,
+  PayloadFormatVersion,
+} from '../integration';
+import { HttpMethod } from '../route';
+import { IVpcLink } from '../vpc-link';
+
+/**
+ * Properties to initialize a new `HttpProxyPrivateIntegration`.
+ */
+export interface HttpProxyPrivateIntegrationProps {
+  /**
+   * The vpc link to be used for the private integration
+   */
+  readonly vpcLink: IVpcLink;
```

Comment:
Can we make this optional and create a new `VpcLink` as part of the `bind()` operation and relieve the user from having to create this?

The list of VpcLinks created per vpc could be cached in the `HttpApi` construct. If another integration from the same vpc is introduced, reuse the vpcLink that was already created.

The above logic will be skipped if the optional `vpcLink` property is specified.

Reply:
The above one will be a tricky implementation since the current modelling of integration themselves are not as constructs. So we will have to create the `VpcLink` in the `Route` construct where the `bind` method of integrations are called. 

We might then also have to touch the defaultIntegration and HttpApi for the new Route behaviour and would need to see how we do this for non-private integration. Therefore I have decided to go ahead without this change for now and we could take it as part of further enhancements.